src/codeql-setup.ts

```diff
@@ -0,0 +1,167 @@
+ */
+export async function setupCodeQLBundle(
+  tempDir: string,
+  cliVersion: string,
+): Promise<string> {
+  const source = getCodeQLSource(cliVersion);
+
+  let codeqlFolder: string;
+  switch (source.sourceType) {
+    case "toolcache":
+      codeqlFolder = source.codeqlFolder;
+      debug(`CodeQL found in cache ${codeqlFolder}`);
+      break;
+    case "download": {
+      codeqlFolder = await downloadCodeQL(
+        cliVersion,
+        source.codeqlURL,
+        tempDir,
+      );
+      break;
+    }
+    default:
+      assertNever(source);
+  }
+
+  return codeqlFolder;
+}
+
+/**
+ * Determine where to find the CodeQL tools. This will check the tool cache
+ * first, and if the tools are not found there, it will provide a download
+ * URL for the tools.
+ *
+ * @param cliVersion The CLI version of the CodeQL bundle to find
+ */
+function getCodeQLSource(cliVersion: string): CodeQLToolsSource {
+  // If we find the specified CLI version, we always use that.
+  const codeqlFolder = findInToolCache("CodeQL", cliVersion);
+
+  if (codeqlFolder) {
+    info(`Found CodeQL tools version ${cliVersion} in the toolcache.`);
+
+    return {
+      codeqlFolder,
+      sourceType: "toolcache",
+    };
+  }
+
+  info(`Did not find CodeQL tools version ${cliVersion} in the toolcache.`);
+
+  /** Tag name of the CodeQL bundle, for example `codeql-bundle-v2.17.1`. */
+  const tagName = `codeql-bundle-v${cliVersion}`;
+
+  const url = `https://github.com/github/codeql-action/releases/download/${tagName}/${getCodeQLBundleName()}`;
+
+  return {
+    codeqlURL: url,
+    sourceType: "download",
+  };
+}
+
+/**
+ * @param cliVersion The CLI version of the CodeQL bundle to download
+ * @param codeqlURL The URL to download the CodeQL bundle from
+ * @param tempDir The temporary directory to download the CodeQL bundle to
+ * @return the path to the downloaded CodeQL tools folder
+ */
+async function downloadCodeQL(
+  cliVersion: string,
+  codeqlURL: string,
+  tempDir: string,
+): Promise<string> {
+  const headers: OutgoingHttpHeaders = {
+    accept: "application/octet-stream",
+  };
+  info(`Downloading CodeQL tools from ${codeqlURL} . This may take a while.`);
+
+  const dest = join(tempDir, uuidV4());
+  const finalHeaders = Object.assign(
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    { "User-Agent": "CodeQL Variant Analysis Action" },
+    headers,
+  );
+
+  const toolsDownloadStart = performance.now();
+  const archivedBundlePath = await downloadTool(
+    codeqlURL,
+    dest,
+    undefined,
+    finalHeaders,
+  );
+  const toolsDownloadDurationMs = Math.round(
+    performance.now() - toolsDownloadStart,
+  );
+
+  debug(
+    `Finished downloading CodeQL bundle to ${archivedBundlePath} (${toolsDownloadDurationMs} ms).`,
+  );
+
+  debug("Extracting CodeQL bundle.");
+  const extractionStart = performance.now();
+  const extractedBundlePath = await extractTar(archivedBundlePath);
```

Comment:
I think that's because this repository has an `ACTIONS_STEP_DEBUG` secret defined at the repository level, so it always has debug logging turned on which [enables the `-v` flag](https://github.com/actions/toolkit/blob/ae38557bb0dba824cdda26ce787bd6b66cf07a83/packages/tool-cache/src/tool-cache.ts#L257-L259). Usually, you wouldn't see this.